streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java

```diff
@@ -81,14 +81,19 @@ public StreamsBuilder() {
         internalStreamsBuilder = new InternalStreamsBuilder(internalTopologyBuilder);
     }
 
-    protected StreamsBuilder(final TopologyConfig topologyConfigs) {
+    /**
+     * Create a {@code StreamsBuilder} instance.
+     *
+     * @param topologyConfigs    the streams configs that apply at the topology level. Please refer to {@link TopologyConfig} for more detail
+     */
+    public StreamsBuilder(final TopologyConfig topologyConfigs) {
         topology = getNewTopology(topologyConfigs);
         internalTopologyBuilder = topology.internalTopologyBuilder;
         internalStreamsBuilder = new InternalStreamsBuilder(internalTopologyBuilder);
     }
 
     protected Topology getNewTopology(final TopologyConfig topologyConfigs) {
```

Comment:
Not specific to this PR since it looks like we already have the TopologyConfig class as a parameter here, but if it's going to be in the public API then we need to move it out of the current package ie the `internals.namedtopology` one. Because it's definitely a public API if it's going to be a parameter of some other public API, and we don't want users to think that everything under the `internals.namedtopology` is also fair game since we haven't done the KIP for it yet

cc @guozhangwang @showuon @wcarlson5

Reply:
@ableegoldman , just to confirm, you mean [this constructor](https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java#L72-L75), right? 
If so, I agree we should remove it after we introduced the public API in `StreamBuilder`.

Reply:
@ableegoldman , Sorry, I don't understand what you mean here. Are you referring this [newNamedTopologyBuilder](https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/processor/internals/namedtopology/KafkaStreamsNamedTopologyWrapper.java#L116-L129) method? Or should I open another JIRA ticket for it?

Reply:
I think the point is that `TopologyConfig` is currently still in `org.apache.kafka.streams.processor.internals.namedtopology`, which is not considered a public API. So for the public class `StreamsBuilder` to take it as a param, we need to move `TopologyConfig` out of that package in this PR, potentially to `org.apache.kafka.streams` as with `StreamsConfig` and `StreamsBuilder` classes.

I thought this is included as part of KIP-591 to make `TopologyConfig` as a public API (note in [KAFKA-13281](https://issues.apache.org/jira/browse/KAFKA-13281), we did not make it a public API yet), as "Currently, the internal implementation for topology-config is all set. In this KIP, we'll publish the API to allow pass the topology-config into StreamBuilder ." already.

Reply:
Yes, agree with you. This should be part of KIP-591 to make the `TopologyConfig` as public API. Moved out of `internals.namedtopology` package now. Thanks.

Reply:
@mjsax it's used by the named topology feature -- there is a  `NamedTopologyBuilder` class that extends `StreamsBuilder` and overrides this method to return a `NamedTopology`.

I know it's a bit awkward, I'll be able to clean it up once I've done the KIP for this feature (which is, finally, in progress)